code/game/objects/items/weapons/storage/wallets.dm

```diff
@@ -30,20 +30,34 @@
 	slot_flags = SLOT_ID
 
 	var/obj/item/weapon/card/id/front_id = null
+	var/obj/item/weapon/card/id/net_id = null
 
 
-/obj/item/weapon/storage/wallet/remove_from_storage(obj/item/W, atom/new_location)
+/obj/item/weapon/storage/wallet/New()
+	..()
+	net_id = new /obj/item/weapon/card/id(src)
+
+/obj/item/weapon/storage/wallet/remove_from_storage(obj/item/W as obj, atom/new_location)
 	. = ..(W, new_location)
 	if(.)
 		if(W == front_id)
 			front_id = null
+		if(istype(W, /obj/item/weapon/card/id))
+			refreshID()
 			update_icon()

```

Comment:
you changed the spot where update_icon() is called, now it will always update regardless if the ID is in front